web-runtime/src/marine_js.rs

```diff
@@ -0,0 +1,305 @@
+                    Some(Export::Function(ProcessedExport {
+                        sig,
+                        name: export.name.to_string(),
+                    }))
+                } else {
+                    crate::js_log(&format!("it is not a function"));
+                    None
+                }
+            })
+            .collect::<Vec<Export>>();
+        exports.push(Export::Memory);
+        crate::js_log(&format!("processed exports"));
+        Self {
+            exports,
+            module_name,
+        }
+    }
+
+    pub fn get(&self, name: &str) -> Result<DynFunc<'_>, String> {
+        crate::js_log(&format!("Exports.get called with name {}", name));
+        let export = self.exports.iter().find(|export| {
+            if let Export::Function(func) = export {
+                func.name == name
+            } else {
+                false
+            }
+        });
+        match export {
+            Some(Export::Function(function)) => Ok(DynFunc {
+                signature: function.sig.clone(),
+                name: function.name.clone(),
+                module_name: self.module_name.clone(),
+                data3: Default::default(),
+            }),
+            Some(_) | None => Err(format!("cannot find export {}", name)),
+        }
+    }
+}
+
+#[derive(Clone)]
+pub struct ProcessedExport {
+    sig: FuncSig,
+    name: String,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone, Copy, PartialEq, Eq, Hash)]
+pub enum WType {
+    /// The `i32` type.
+    I32,
+    /// The `i64` type.
+    I64,
+    /// The `f32` type.
+    F32,
+    /// The `f64` type.
+    F64,
+    /// The `v128` type.
+    V128,
+}
+
+impl std::fmt::Display for WType {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        write!(f, "{:?}", self)
+    }
+}
+
+/// Represents a WebAssembly value.
+///
+/// As the number of types in WebAssembly expand,
+/// this structure will expand as well.
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq)]
+pub enum WValue {
+    /// The `i32` type.
+    I32(i32),
+    /// The `i64` type.
+    I64(i64),
+    /// The `f32` type.
+    F32(f32),
+    /// The `f64` type.
+    F64(f64),
+    /// The `v128` type.
+    V128(u128),
+}
+
+/// An iterator to an instance's exports.
+pub struct ExportIter<'a> {
+    //_data: PhantomData<&'a i32>,
+    exports: &'a Exports,
+    index: usize,
+}
+
+impl<'a> ExportIter<'a> {
+    pub(crate) fn new(exports: &'a Exports) -> Self {
+        Self {
+            //_data: PhantomData::<&'a i32>::default(),
+            exports,
+            index: 0,
+        }
+    }
+}
+
+impl<'a> Iterator for ExportIter<'a> {
+    type Item = (String, Export);
+    fn next(&mut self) -> Option<(String, Export)> {
+        let export = self.exports.exports.get(self.index);
+        self.index += 1;
+        export.map(|export| (export.name(), export.clone()))
+    }
+}
+
+#[derive(Clone)]
+pub struct WasmMemory {
```

Comment:
We have a lot of memory objects, so it's better to choose names very carefully. Here this looks like as a proxy-js object, maybe name it `JsWasmMemoryProxy`?

Reply:
good point, will use this name